datarobot_batch_scoring/batch_scoring.py

```diff
@@ -78,29 +78,61 @@ def slow_to_csv_chunk(data, header):
     return buf.getvalue()
 
 
-class FastReader(object):
-    """A reader that only reads the file in text mode but not parses it. """
+class Recoder:
+    """
+    Iterator that reads an encoded stream and decodes the input to UTF-8
+    for Python 2. In Python 3 the open function decodes the file.
+    """
+    def __init__(self, f, encoding):
+        f.seek(0)
+        # self.reader = codecs.getreader(encoding)(f)
+        if six.PY3:
+            self.reader = f
+        else:
+            self.reader = codecs.StreamRecoder(f, codecs.getencoder('utf-8'),
+                                               codecs.getdecoder('utf-8'),
+                                               codecs.getreader(encoding),
+                                               codecs.getwriter(encoding))
+
+    def __iter__(self):
+        return self
+
+    def next(self):   # python 3
```

Comment:
def __next__(self):
    return next(self.reader)

should be completely compatible for py2/3